test/tests/helper.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const app = require('../../server');
+const chai = require('chai');
+const db = require('../../db');
+const mongoose = require('mongoose');
+const clearDB  = require('mocha-mongoose')(db.uri);
+const request = require('supertest');
+
+chai.use(require('chai-as-promised'));
+global.expect = chai.expect();
+global.should = chai.should();
+global.request = request(app);
+
+
+
+beforeEach((done, err) => {
+  if(mongoose.connection.db) return done(err);
+
+  mongoose.connect(dbURI);
```

Comment:
`db.uri`

Reply:
`mongoose.connect(...)` is async, we need to wait for it.
http://mongoosejs.com/docs/api.html#index_Mongoose-connect